libsolidity/ast/Types.cpp

```diff
@@ -2168,11 +2178,15 @@ std::vector<std::tuple<VariableDeclaration const*, u256, unsigned>> ContractType
 		for (VariableDeclaration const* variable: contract->stateVariables())
```

Comment:
One thing that still somewhat bothers me here is that there's nothing preventing someone from running `ContractType::stateVariables()` on any contract in the inheritance hierarchy, even though you'll only get a valid result if it's the top level one.

I checked the uses and looks like we fortunately never do this. Still, I think it might be a good idea to rename the function to something like `linearizedStateVariables()` or even `variableLayout()` to make it clearer. Currently it's too easily confused with `ContractDefinition::stateVariables()` which *is* valid for any contract.

The docstring should also say that the function works like this and that the returned slots are absolute.